src/algorithm/poly-partition.ts

```diff
@@ -0,0 +1,479 @@
+import {Polygon} from "../polygon";
+import {Vector} from "../vector";
+import {isConvex} from "./convex";
+
+export enum Orientation {
+    CCW = 1,
+    CW = -1,
+    UNORDERED = 0,
+}
+
+/**
+ * Polygon used in poly partition.
+ * 1. It is a polygon.
+ * 2. Can be either a normal polygon or a hole in outer polygon.
+ * 3. Points are sorted in counter-clockwise order from it's center when it's a normal polygon, or clockwise if it's a hole.
+ */
+export class PartitionPolygon extends Polygon {
```

Comment:
Do you think composition is better here? because we probably can reuse the polygon generated from the map loader later.
ex.
class PartitionPolygon {
  Polygon polygon;
}